unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/GodMode/CatalogBtnView.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public interface ICatalogBtnView
+{
+    event Action OnCatalogButtonClick;
+    event Action OnHideTooltip;
+    event Action<BaseEventData, string> OnShowTooltip;
+
+    void ConfigureEventTrigger(EventTriggerType eventType, UnityAction<BaseEventData> call);
+    void OnPointerClick();
+    void OnPointerEnter(PointerEventData eventData);
+    void OnPointerExit();
+    void RemoveEventTrigger(EventTriggerType eventType);
+}
+
+public class CatalogBtnView : MonoBehaviour, ICatalogBtnView
+{
+    public event Action OnCatalogButtonClick;
+    public event Action<BaseEventData, string> OnShowTooltip;
+    public event Action OnHideTooltip;
+
+    [SerializeField] internal Button mainButton;
+    [SerializeField] internal string tooltipText = "Open Catalog (C)";
+    [SerializeField] internal EventTrigger catalogButtonEventTrigger;
+    [SerializeField] internal InputAction_Trigger toggleCatalogInputAction;
+
+    private const string VIEW_PATH = "GodMode/CatalogBtnView";
+
+    internal static CatalogBtnView Create()
+    {
+        var view = Instantiate(Resources.Load<GameObject>(VIEW_PATH)).GetComponent<CatalogBtnView>();
+        view.gameObject.name = "_CatalogBtnView";
+
+        return view;
+    }
+
+    private void Awake()
+    {
+        mainButton.onClick.AddListener(OnPointerClick);
+        toggleCatalogInputAction.OnTriggered += (action) => OnPointerClick();
+        ConfigureEventTrigger(EventTriggerType.PointerEnter, (eventData) => OnPointerEnter((PointerEventData)eventData));
+        ConfigureEventTrigger(EventTriggerType.PointerExit, (eventData) => OnPointerExit());
+    }
+
+    private void OnDestroy()
+    {
+        mainButton.onClick.RemoveListener(OnPointerClick);
+        toggleCatalogInputAction.OnTriggered -= (action) => OnPointerClick();
+        RemoveEventTrigger(EventTriggerType.PointerEnter);
+        RemoveEventTrigger(EventTriggerType.PointerExit);
+    }
+
+    public void ConfigureEventTrigger(EventTriggerType eventType, UnityAction<BaseEventData> call)
+    {
+        EventTrigger.Entry entry = new EventTrigger.Entry();
+        entry.eventID = eventType;
+        entry.callback.AddListener(call);
+        catalogButtonEventTrigger.triggers.Add(entry);
+    }
```

Comment:
We have this function in a lot of views, should we move this function to `BuilderInWorldUtils` with another param for the event trigger?

Reply:
You're completely right. I have created both functions `ConfigureEventTrigger()` and `RemoveEventTrigger()` in `BuilderInWorldUtils`. You can see the cahnge in https://github.com/decentraland/explorer/pull/2103/commits/72fa0958c890bf871abd1934fed1fd8bf6db46f1. Thanks for the suggestion!